File: 백준/Silver/2725. 보이는 점의 개수/보이는 점의 개수.py
c = int(input())
#기울기가 달라야함

def get_gcd(a,b):
    while a % b != 0:
        a, b = b, a % b
    return b

arr = [0] * 1001
arr[1] = 3

for i in range(2, 1001):
    cnt = 0
    for j in range(1, i + 1):
        if get_gcd(i, j) == 1:
            cnt += 1
    arr[i] = arr[i - 1] + cnt * 2

for _ in range(c):
    print(arr[int(input())])

File: 백준/Silver/2725. 보이는 점의 개수/보이는 점의 개수.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        int c = Integer.parseInt(br.readLine());
        int[] arr = new int[1001];
        arr[1] = 3;

        for (int i = 2; i < 1001; i++) {
            int cnt = 0;
            for (int j = 1; j < i + 1; j++) {
                if(gcd(Math.max(i, j), Math.min(i, j)) == 1){
                    cnt++;
                }
            }
            arr[i] = arr[i - 1] + cnt * 2;
        }
        
        for (int t = 0; t < c; t++) {
            int n = Integer.parseInt(br.readLine());
            sb.append(arr[n] + "\n");
        }
        System.out.println(sb);
    }

    public static int gcd(int a, int b){
        while(a % b != 0){
            int tmp = a;
            a = b;
            b = tmp % b;
        }
        return b;
    }
}